### ISSUE TITLE
Convex Dominance in Deep Learning I: A Scaling Law of Loss and Learning Rate

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#502]
## Links
https://www.arxiv.org/abs/2602.07145

## 論文概要  
**Convex Dominance in Deep Learning I: A Scaling Law of Loss and Learning Rate**  
Zhiqi Bu et al., 2026 

---

### ■ 3行要約

- 本論文は、深層学習の非凸最適化が実際には「凸的に振る舞う」ことに着目し、学習率と損失の関係を理論・実証の両面から明らかにする。  
- 特定の学習率スケジュールでは、最終損失が **O(1/√T)** で収束し、最適学習率も **O(1/√T)** に比例することを示す。  
- これに基づき、訓練ステップ数（T）とモデルサイズ（N）にまたがる「二次元スケーリング則」を構築し、大規模モデルへ最大80倍以上の外挿が可能であることを実証した。

---

# 1. 研究背景と問題設定

深層学習は理論的には非凸最適化問題であり、一般には解析困難とされている。しかし実際の学習曲線は、驚くほど凸最適化理論と整合的な挙動を示すことが知られている。

本研究は以下の問いを立てる：

> 深層学習の損失ダイナミクスは、凸最適化理論の枠組みでどこまで説明できるのか？

特に焦点は：

- 学習率スケジュールと損失収束速度の関係
- 最適学習率のスケーリング則
- モデルサイズ・訓練長に跨る予測可能性

---

# 2. 理論的出発点：凸SGDの損失上界

論文はまず、凸損失 + 有界勾配条件下でのSGDの損失上界を再整理する（第2章）。

学習率を

- peak learning rate：η_peak  
- スケジュール関数：s_t(T)

と分解し、

> 学習率列 {η_t} → 最終損失上界

という明示的対応関係を導く。

---

## 2.1 重要な理論結果（Table 1, p.4）

各スケジュールに対し、最終損失は以下の形になる：

L(T) ≈ L* + C₁ / (T η_peak) + C₂ η_peak

これを最小化すると：

- 最適学習率：η_peak ∝ 1/√T  
- 最終損失：L(T) − L* ∝ 1/√T  

ただし重要なのは：

- **線形減衰・cosine減衰・WSDのみが最適収束を達成**
- 定数学習率は log(T)/T に悪化

---

## 2.2 新規性①：学習率スケジュールの「資格試験」

著者らは Condition 2.5 にて、

> 学習率スケジュールが O(1/√T) を達成できるかどうかを  
> 訓練なしで判定する条件

を提示する。

### Figure 1（p.5）

- 左：線形減衰 → 1/√Tに対し完全な直線（R²=1）
- 中央：定数 → 不適合
- 右：log(T)/T に比例する別スケーリング

→ 理論予測と完全一致

これは従来研究（Schaipp et al., 2025）を拡張した点であり、任意スケジュールに適用可能である。

---

# 3. 深層学習への一般化（最大の貢献）

ここが本論文の核心。

理論は凸SGD限定だが、著者は次を実証：

> 非凸深層学習 + AdamW + Muon + LoRA  
> においても同じ構造が成立

---

## 3.1 Generalization 1（式3.1）

最終損失は

L(τ) ≈ L̃∞ + D̃² / Ση_t + G̃² ×（学習率関数項）

という**抽象形**で表現できると提案。

ここで：

- D̃, G̃ は理論的定数ではなく「データ駆動で推定」
- L̃∞ は不可約損失

---

### Figure 2–5（p.6–7）

ResNet / GPT2 / AdamW / Muon で

- 前半でフィット
- 後半を予測

R² ≥ 0.95

特に：

- WSD → 学習率減衰と同時に損失が急減
- Cyclic → 周期振動を完全再現

→ **sequence-to-sequence予測成功**

これが従来研究との差分：

- Schaippらは1モデルのみ
- 本研究は多モデル・多オプティマイザで実証

---

# 4. 最終損失の漸近形（Generalization 2,3）

適格スケジュールでは最終損失は

L(T) ≈ L̃∞ + q̃₁/(T η_peak) + q̃₂ η_peak

最適化すると

L(T) ≈ L̃∞ + 2√(q̃₁ q̃₂) / √T

---

## Figure 6（p.8）

Li et al. 2025 の大規模LLM結果を再解析：

- 学習率 vs 損失
- 密モデル・MoE両方でR² ≥ 0.95

---

## Figure 7（p.8）

Chinchilla実験の再構築データ：

- 0.074B 〜 12.56B
- FLOPs最大 1e22
- 最大300Bトークン

すべて

損失 vs 1/√T が直線

誤差1%以内

→ 大規模LLMにおいても成立

---

# 5. 二次元スケーリング則（本論文の最重要貢献）

著者は以下を提示：

L(N,T) ≈ L̃∞(N) + Q̃(N) / √T  
η_peak(N,T) ∝ 1/√T

---

## 5.1 実験設定

GPT2：

- 0.1B
- 1B
- 7B
- 240実験

---

## Figure 8–9（p.10）

0.1Bで最適 η_ref を推定：

- AdamW → 0.3
- Muon → 10

→ 80倍の訓練長に外挿成功

---

## Figure 10（p.10）

モデルサイズ跨ぎ（0.1B→7B）：

- 同じ η_ref を転移
- 70倍外挿成功

---

## Figure 11（p.11）

マルチモーダルVLMでも成立

---

# 6. 従来研究との違い

| 既存研究 | 本研究の差分 |
|----------|-------------|
| Kaplan/Hoffmann | 損失のみスケール |
| Bi/Bjorck | 学習率のみスケール |
| muP | モデルサイズ転移のみ |

本研究：

> 損失と学習率を同時に予測する統一理論

さらに：

- 訓練なしでスケジュール適格判定
- 80× horizon extrapolation
- 70× model size extrapolation

---

# 7. 実証方法の特徴

- データ駆動線形回帰
- 1/√T に対する直線性を評価
- R² ほぼ 1.0

アブレーション（Table 7, p.24）：

- weight decay
- batch size
- clipping
- seed

すべてで成立

---

# 8. 制限事項

Figure 12（p.23）：

- 過学習が強い場合
  - train loss → 成立
  - test loss → 破綻

→ 汎化理論は未解明

---

# 9. 本論文の本質的インパクト

この研究は次を示した：

1. 深層学習は早期に「弱凸的支配状態」に入る
2. 学習率は 1/√T にスケールすべき
3. 損失は 1/√T で減少
4. 小規模実験から大規模設定へ予測可能

---

# 10. まとめ

本論文は、

> 「深層学習の非凸最適化は、実用上ほぼ凸的に振る舞う」

という強い実証的主張を提示し、

- 理論（凸SGD）
- 実証（ResNet〜GPT〜MoE）
- スケーリング則（T, N）

を統合した。

これはハイパーパラメータ探索のコスト削減と、
大規模学習設計に極めて重要な示唆を与える研究である。

---

<img width="1145" height="238" alt="Image" src="https://github.com/user-attachments/assets/9d1483ea-ea46-4bba-8c40-2973edac716a" />

<img width="1511" height="393" alt="Image" src="https://github.com/user-attachments/assets/a1314b58-664d-4047-9a57-c9ae8ce0e499" />

<img width="1148" height="347" alt="Image" src="https://github.com/user-attachments/assets/7a30db0e-3280-49d6-b92d-f405a1c023ef" />
